packages/graphql-language-service-utils/src/file.ts

```diff
@@ -0,0 +1,81 @@
+import fs from 'fs';
+
+export function getFileExtension(filePath: string): string | null {
+  const pathParts = /^.+\.([^.]+)$/.exec(filePath);
+  // if there's a file extension
+  if (pathParts && pathParts.length > 1) {
+    return pathParts[1];
+  }
+  return null;
+}
+
+export function getPathWithoutExtension(
+  filePath: string,
+  extension: string | null,
+) {
+  let pathWithoutExtension = filePath;
+  if (extension) {
+    pathWithoutExtension = filePath.substr(
+      0,
+      filePath.length - (extension.length + 1),
+    );
+  }
+  return pathWithoutExtension;
+}
+
+// these make webpack happy
+
+export function resolveFile(filePath: string) {
+  const extension = getFileExtension(filePath);
+  const pathWithoutExtension = getPathWithoutExtension(filePath, extension);
+  switch (extension) {
+    case 'js': {
+      return require.resolve(pathWithoutExtension + '.js');
+    }
+    case 'json': {
+      return require.resolve(pathWithoutExtension + '.json');
+    }
+    default: {
+      if (fs.existsSync(filePath + `.js`)) {
+        return require.resolve(filePath + '.js');
```

Comment:
I'm surprised we can bundle with these `fs.existsSync` calls?

Reply:
Would it be better to use a `try` block and skip the `fs` calls?

I don't like the idea of `fs.existsSync` being called after startup either; can we memoize this function?

Reply:
good idea

Reply:
do you mean like this?

```ts
export function resolveFile(filePath: string) {
  const extension = getFileExtension(filePath);
  const pathWithoutExtension = getPathWithoutExtension(filePath, extension);
  switch (extension) {
    case 'js': {
      return require.resolve(pathWithoutExtension + '.js');
    }
    case 'json': {
      return require.resolve(pathWithoutExtension + '.json');
    }
    default: {
      try {
        require.resolve(filePath + '.js')
      }
      catch (err) {
        try {
          require.resolve(filePath + '.json')
        }
        catch (err) {
          if (extension) {
            throw Error(
              `cannot require.resolve() module with extension '${extension}'`,
            );
          }
        }
      }
    }
  }
}
```